mm2src/mm2_main/src/lp_swap.rs

```diff
@@ -654,14 +654,12 @@ pub fn get_locked_amount(ctx: &MmArc, coin: &str) -> MmNumber {
     locked
 }
 
-/// Get number of currently running swaps
-pub fn running_swaps_num(ctx: &MmArc) -> u64 {
+/// Clear up all the running swaps.
+///
+/// This also auto-stops any running swaps since their abort handles will get triggered (assuming these abort handles aren't already triggered by other means).
+pub fn clear_running_swaps(ctx: &MmArc) {
     let swap_ctx = SwapsContext::from_ctx(ctx).unwrap();
-    let swaps = swap_ctx.running_swaps.lock().unwrap();
-    swaps.iter().fold(0, |total, swap| match swap.upgrade() {
-        Some(_) => total + 1,
-        None => total,
-    })
+    swap_ctx.running_swaps.lock().unwrap().clear();
```

Comment:
Does clearing the map stops swap processes?

Reply:
yup. the store of `running_swaps` stores the swaps (`AtomicSwap` object) along with the abort handles (which will trigger on `Drop`) they are running on.

Reply:
Can you please mention that on the code?

Reply:
https://github.com/KomodoPlatform/komodo-defi-framework/blob/dd5418fc2444679a57acdb5370788b2ac1fcc1da/mm2src/mm2_main/src/lp_swap.rs#L657-L660

it's already in the doc comment. or u mean somewhere else?

Reply:
Ah didn't see that. I was referring to caller side but it's fine

Reply:
big ooops:

> yup. the store of running_swaps stores the swaps (AtomicSwap object) along with the abort handles (which will trigger on Drop) they are running on.

this is completely wrong. this feat is in another PR. i mixed things up.

no, clearing the running swaps will not stop these swaps. (though could be stopped via other means, e.g. `abort_all` from a bigger subsystem).

Reply:
Ok that is def. worth to mention that on the line.

Reply:
I tried this code, looks like clear_running_swaps fn is needed to clear swaps_ctx in MmCtx to allow to release ctx so stop_and_wait_for_ctx_is_dropped fn can finish (used in wasm tests).

(this note to memorise)

Reply:
updated the doc comment accordingly till the other PR that stores the abort handle lands in.